src/style_io.py

```diff
@@ -133,3 +137,69 @@ def generate_preview(style: RcParams) -> Gio.File:
         axis.set_xlabel(_("Y Label"))
         figure.savefig(wrapper, format="svg")
     return file
+
+
+def generate_system_preview(light_style: RcParams,
+                            dark_style: RcParams) -> Gio.File:
+    # Generate light variant
+    with rc_context(light_style):
+        figure = Figure(figsize=(5, 3))
+        axis = figure.add_subplot()
+        axis.spines.bottom.set_visible(True)
+        axis.spines.left.set_visible(True)
+        if not light_style["axes.spines.top"]:
+            axis.tick_params(which="both", top=False, right=False)
+        axis.plot(_PREVIEW_XDATA, _PREVIEW_YDATA1)
+        axis.plot(_PREVIEW_XDATA, _PREVIEW_YDATA2)
+        axis.set_xlabel(_("X Label"))
+        axis.set_ylabel(_("Y Label"))
+        canvas = FigureCanvas(figure)
```

Comment:
Is creating a canvas really needed here? shouldn't `figure.savefig()` work as well? (afaik the canvas implicitly calls this either way)